src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java

```diff
@@ -94,11 +95,12 @@ public class User implements Serializable {
     @ElementCollection
     private Set<Long> likedByUsers = new TreeSet<>();
 
-    @ElementCollection
-    private Set<Long> matches = new TreeSet<>();
+    // cascade must include remove, so that the matches are deleted as well (important)
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "users")
+    private Set<Match> matches = new HashSet<>();
```

Comment:
is fetch type not per default lazy? else i would say we have to change that everywere. as lazy uses less memory normaly (only if needed)

Reply:
Yes it is, but writing it here makes it crystal clear for everyone